packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -681,21 +684,40 @@ class DropdownMenuItem<T> extends _DropdownMenuItemContainer {
   /// Creates an item for a dropdown menu.
   ///
   /// The [child] argument is required.
+  ///
+  /// If [enabled] is `false`, [value] and [onTap] should be null.
   const DropdownMenuItem({
     Key? key,
     this.onTap,
     this.value,
+    this.enabled = true,
     required Widget child,
   }) : assert(child != null),
+       assert(enabled || (!enabled && value == null), 'if enabled is false, value should be null'),
+       assert(enabled || (!enabled && onTap == null), 'if enabled is false, onTap should be null'),
        super(key: key, child: child);
 
   /// Called when the dropdown menu item is tapped.
+  ///
+  /// If [enabled] is `false`, this property should be `null`.
```

Comment:
Actually, in keeping with our other widgets, this should probably be what determines if it is enabled or not.  We typically disable items (buttons, etc.) when their onTap is null. That way you don't need a separate "enabled" accessor: it is sufficient to just set onTap to null.

Reply:
One problem is that this would be a breaking change -- Currently, the way the `DropdownMenuItem` works is that defining `onTap` is a callback that is called on top of the behavior of selecting the item. This means that any user that hasn't defined their `onTap` callback will suddenly have their menu items be disabled.

Reply:
Considering the present implementation (on master branch) of `DropdownMenuItem`, `onTap` can be `null` and the item will still be enabled. Altering this code design (according to your suggestion) will be a breaking change.

Also, less important, but there may be a situation where the user doesn't want an `onTap` call on an enabled item. In this case the user is compelled to add a `VoidCallback` to ensure the item is enabled.